### DISCUSSION TITLE
V14.6, blink, friendly snippets and extras

### DISCUSSION THREAD

**Discussion by resistorsoftware** [LazyVim/LazyVim#5250] [General]
I have nice snippets in my Ruby files such that I see them pop-up via *blink* since LazyVim is not using Luasnip. The friendly snippets repository speaks of adding framework extras such as Rails via a custom Luasnip modification. 
```
For example: if you want to add rails snippets to ruby.

With LuaSnip:

require'luasnip'.filetype_extend("ruby", {"rails"})
```
Since this incarnation of LazyVim is not using Luasnip, is there anyone kind enough and experienced enough with Lua and Neovim and LazyVim to point me in the right direction to adding this extension to *Blink* so that I would gain these Rails snippets in LazyVim 14.6++ moving forward?

Is it as simple as enabling Luasnip from LazyExtras and then adding that configuration?

**Comment by surmish** [comment]
I do something like this to associate file extensions with filetypes (in my `autocmds.lua`):

```
 -- File type setting
 local set_filetype = function(pattern, filetype)
   vim.api.nvim_create_autocmd({ "VimEnter", "BufRead", "BufNewFile", "BufEnter" }, {
     pattern = pattern,
     callback = function()
       vim.api.nvim_set_option_value("filetype", filetype, { scope = "local" })
     end,
   })
 end

# Use it as follows:
set_filetype({ "*.mk", "Make*" }, "make")
```

**Reply by resistorsoftware** [reply]
It's not associating file extensions so much as getting blink to show the Ruby and Rails snippets when those special keywords are inserted. Blink comes with Ruby snippets working, but to add Rails requires an extra load as in the Luasnip code I pointed out. So my noobie problem is not understanding whether this should be converted to a blink *without* luasnip or blink *with* luasnip.
